pkg/proxy/proxy.go

```diff
@@ -258,6 +259,19 @@ func (p *Proxy) addUserContext() echo.MiddlewareFunc {
 	}
 }
 
+func (p *Proxy) stripInvalidHeaders() echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(ctx echo.Context) error {
+			for header := range ctx.Request().Header {
+				if strings.HasPrefix(header, "Impersonate-") {
+					ctx.Request().Header.Del(header)
```

Comment:
Let's log this. In case we want to check in the future if anyone tried to set the impersonate headers ;)